### PR TITLE
Add windows compatibility and support type, media and crossorigin attributes

### PR DESCRIPTION
**Pull request by andrevandal**
According to [Netlify post](https://www.netlify.com/blog/2017/07/18/http/2-server-push-on-netlify/), link path should be relative of root folder and Fonts [should have](https://stackoverflow.com/questions/46401942/http2-pushed-webfonts-not-used) `crossorigin` tag.

### CODE REVIEW

**Comment by jmblog** [comment]
Hi @derevandal, thanks for your PR.

1. Removing `${generateDir}`

I think that it doesn't work well if removing `${generateDir}`.
How do you write the `modules` section of your `nuxt.config.js`? A path in `resources` should be the relative to the `dist` directory.

```js
{
  modules: [
    'nuxt-netlify-http2-server-push',
    {
      resources: [
        // Use the relative path to the dist directory
        { path: '**/*.js', as: 'script' },

       // DON'T DO THIS
       // { path: 'dist/**/*.js, as: 'script' }
      ]
    }
  ];
}
```

2. Adding `crossorigin` feature

That's the nice suggestion!

According to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#attr-crossorigin, a `crossorigin` attribute accepts two values (`anonymous` or `use-credentials`).

And a `type` and `media` attributes also should be acceptable. (Ref: https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content)

Example of `nuxt.config.js`:
```js
{
  modules: [
    'nuxt-netlify-http2-server-push',
    {
      resources: [
        { path: '**/*.js', as: 'script' },
        { path: 'fonts/*.woff2', as: 'font', type: 'font/woff2', crossorigin: 'anonymous' },
        { path: 'images/bg-image-narrow.png', as: 'image', media: '(max-width: 600px)', },
      ]
    }
  ];
}
```

What do you think of that? Could you change your PR? Or I'll add a change.

**Comment by andrevandal** [comment]
Hi, @jmblog, you're welcome! XD

1 .Removing ${generateDir}

When I use `yarn generate`, I get this:
```
/*
  Link: <C:/Users/derevandal/dev/test/dist/assets/007d3d2a8b4f92d77e95.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/0a87762375116af4200c.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/16ec962df1e211684689.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/203fb7acbeb231a1d304.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/2c7f0d10efa6596aaf5b.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/3836b5ba2442f2ce58f7.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/4e3c80d416ba2cf73e0d.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/74929b514537e75b4dac.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/925a1a2e33f1d69738c7.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/9ba58988e489459a84be.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/b81390f7215bfbe05337.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/bf04e2e3ee08a2fa3737.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/assets/workbox.4c4f5ca6.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/sw.js>; rel=preload; as=script
  Link: <C:/Users/derevandal/dev/test/dist/fonts/hinted-Book.woff2>; rel=preload; as=fonts
  Link: <C:/Users/derevandal/dev/test/dist/fonts/hinted-Bold.woff2>; rel=preload; as=fonts
```
And my `resources` is:
```js
resources: [
    { path: '**/*.js', as: 'script' },
    {
      path: 'fonts/hinted-GothamRounded-Book.woff2',
      as: 'fonts',
      crossorigin: true
    },
    {
      path: 'fonts/hinted-GothamRounded-Bold.woff2',
      as: 'fonts',
      crossorigin: true
    }
    // { path: 'images/hero.jpg', as: 'image' }
 ]
```

2. Adding crossorigin feature
I'll do it o/

Thanks.

File: lib/module.js [hunk 1]
```diff
@@ -26,10 +26,14 @@ const generateHeaderFile = (context, resources) => {
   const headersFile = `${generateDir}/_headers`;
   let content = '/*\n';
   resources.forEach(resource => {
-    const files = glob.sync(`${generateDir}/${resource.path}`);
+    const files = glob.sync(`${resource.path}`);
     files.forEach(file => {
       const url = file.replace(generateDir, '');
```
Comment (jmblog) [inline lib/module.js]: @derevandal You use Windows, right?

It seems that this line (line 31) doesn't work as expected on Windows. Could you check the values of `generateDir`, `files` and `url`?

Comment (andrevandal) [reply lib/module.js]: I'll test it and post the log here.
Nevertheless, Netlify doesn't push the assets even with `_headers` automatically generated during `yarn generate`. :(

#edit

And yes, I'm using Windows to dev.

Comment (andrevandal) [reply lib/module.js]: @jmblog these are the values that you asked for:
```sh
generateDir: C:\Users\andre.vandal\dev\virada-tecnospeed\dist
resource: {
  path: '**/*.js',
  as: 'script'
}
file: C:/Users/derevandal/dev/test/dist/assets/0878cfcff900ea74e8df.js
url: C:/Users/derevandal/dev/test/dist/assets/0878cfcff900ea74e8df.js 
file: C:/Users/derevandal/dev/test/dist/assets/224cf76102929e9a955a.js
url: C:/Users/derevandal/dev/test/dist/assets/224cf76102929e9a955a.js 
file: C:/Users/derevandal/dev/test/dist/assets/26ac87831e199474822f.js
url: C:/Users/derevandal/dev/test/dist/assets/26ac87831e199474822f.js 
file: C:/Users/derevandal/dev/test/dist/assets/4de967c7be1170f297bc.js
url: C:/Users/derevandal/dev/test/dist/assets/4de967c7be1170f297bc.js 
file: C:/Users/derevandal/dev/test/dist/assets/5bb713b0e5d125922aa4.js
url: C:/Users/derevandal/dev/test/dist/assets/5bb713b0e5d125922aa4.js 
file: C:/Users/derevandal/dev/test/dist/assets/a3dcd05bbe2da6f3be6f.js
url: C:/Users/derevandal/dev/test/dist/assets/a3dcd05bbe2da6f3be6f.js 
file: C:/Users/derevandal/dev/test/dist/assets/a85485e574526399ce46.js
url: C:/Users/derevandal/dev/test/dist/assets/a85485e574526399ce46.js 
file: C:/Users/derevandal/dev/test/dist/assets/b7d7abb2f34e621a0cb9.js
url: C:/Users/derevandal/dev/test/dist/assets/b7d7abb2f34e621a0cb9.js 
file: C:/Users/derevandal/dev/test/dist/assets/c49ca96955f1359e0275.js
url: C:/Users/derevandal/dev/test/dist/assets/c49ca96955f1359e0275.js 
file: C:/Users/derevandal/dev/test/dist/assets/e2c742de3d7699b88134.js
url: C:/Users/derevandal/dev/test/dist/assets/e2c742de3d7699b88134.js 
file: C:/Users/derevandal/dev/test/dist/assets/workbox.4c4f5ca6.js    
url: C:/Users/derevandal/dev/test/dist/assets/workbox.4c4f5ca6.js     
file: C:/Users/derevandal/dev/test/dist/sw.js 
url: C:/Users/derevandal/dev/test/dist/sw.js  
resource: {                                   
  path: '/fonts/hinted-GothamRounded-Book.woff2',
  as: 'fonts',
  crossorigin: true
}
file: C:/Users/derevandal/dev/test/dist/fonts/hinted-GothamRounded-Book.woff2
url: C:/Users/derevandal/dev/test/dist/fonts/hinted-GothamRounded-Book.woff2 
resource: {                                   
  path: '/fonts/hinted-GothamRounded-Bold.woff2',
  as: 'fonts',
  crossorigin: true
}
file: C:/Users/derevandal/dev/test/dist/fonts/hinted-GothamRounded-Bold.woff2
url: C:/Users/derevandal/dev/test/dist/fonts/hinted-GothamRounded-Bold.woff2 
resource: {                                   
  path: '/images/*.svg',
  as: 'image'
}
file: C:/Users/derevandal/dev/test/dist/images/boleto.svg             
url: C:/Users/derevandal/dev/test/dist/images/boleto.svg              
file: C:/Users/derevandal/dev/test/dist/images/cte.svg                
url: C:/Users/derevandal/dev/test/dist/images/cte.svg                 
file: C:/Users/derevandal/dev/test/dist/images/cteos.svg              
url: C:/Users/derevandal/dev/test/dist/images/cteos.svg               
file: C:/Users/derevandal/dev/test/dist/images/dots.svg               
url: C:/Users/derevandal/dev/test/dist/images/dots.svg                
file: C:/Users/derevandal/dev/test/dist/images/esocial.svg            
url: C:/Users/derevandal/dev/test/dist/images/esocial.svg             
file: C:/Users/derevandal/dev/test/dist/images/gnre.svg               
url: C:/Users/derevandal/dev/test/dist/images/gnre.svg                
file: C:/Users/derevandal/dev/test/dist/images/mdfe.svg               
url: C:/Users/derevandal/dev/test/dist/images/mdfe.svg                
file: C:/Users/derevandal/dev/test/dist/images/nfce.svg               
url: C:/Users/derevandal/dev/test/dist/images/nfce.svg                
file: C:/Users/derevandal/dev/test/dist/images/nfe.svg                
url: C:/Users/derevandal/dev/test/dist/images/nfe.svg                 
file: C:/Users/derevandal/dev/test/dist/images/nfse.svg               
url: C:/Users/derevandal/dev/test/dist/images/nfse.svg                
file: C:/Users/derevandal/dev/test/dist/images/notasegura.svg         
url: C:/Users/derevandal/dev/test/dist/images/notasegura.svg          
file: C:/Users/derevandal/dev/test/dist/images/reinf.svg              
url: C:/Users/derevandal/dev/test/dist/images/reinf.svg               
file: C:/Users/derevandal/dev/test/dist/images/sat.svg                
url: C:/Users/derevandal/dev/test/dist/images/sat.svg                 
file: C:/Users/derevandal/dev/test/dist/images/sped.svg               
url: C:/Users/derevandal/dev/test/dist/images/sped.svg                
file: C:/Users/derevandal/dev/test/dist/images/tecnospeed.svg         
url: C:/Users/derevandal/dev/test/dist/images/tecnospeed.svg          
file: C:/Users/derevandal/dev/test/dist/images/virada.svg             
url: C:/Users/derevandal/dev/test/dist/images/virada.svg
```

**Comment by andrevandal** [comment]
I think I resolve this bug on windows.
```sh
start Generating a _headers file... nuxt:netlify-http2-server-push 
generateDir: C:\Users\andre.vandal\dev\virada-tecnospeed\dist
resource: {
  path: '**/*.js',
  as: 'script'
}
file: C:/Users/derevandal/dev/test/dist/assets/0878cfcff900ea74e8df.js
url: /assets/0878cfcff900ea74e8df.js
file: C:/Users/derevandal/dev/test/dist/assets/224cf76102929e9a955a.js
url: /assets/224cf76102929e9a955a.js
file: C:/Users/derevandal/dev/test/dist/assets/26ac87831e199474822f.js
url: /assets/26ac87831e199474822f.js
file: C:/Users/derevandal/dev/test/dist/assets/4de967c7be1170f297bc.js
url: /assets/4de967c7be1170f297bc.js
file: C:/Users/derevandal/dev/test/dist/assets/5bb713b0e5d125922aa4.js
url: /assets/5bb713b0e5d125922aa4.js
file: C:/Users/derevandal/dev/test/dist/assets/a3dcd05bbe2da6f3be6f.js
url: /assets/a3dcd05bbe2da6f3be6f.js
file: C:/Users/derevandal/dev/test/dist/assets/a85485e574526399ce46.js
url: /assets/a85485e574526399ce46.js
file: C:/Users/derevandal/dev/test/dist/assets/b7d7abb2f34e621a0cb9.js
url: /assets/b7d7abb2f34e621a0cb9.js
file: C:/Users/derevandal/dev/test/dist/assets/c49ca96955f1359e0275.js
url: /assets/c49ca96955f1359e0275.js
file: C:/Users/derevandal/dev/test/dist/assets/e2c742de3d7699b88134.js
url: /assets/e2c742de3d7699b88134.js
file: C:/Users/derevandal/dev/test/dist/assets/workbox.4c4f5ca6.js
url: /assets/workbox.4c4f5ca6.js
file: C:/Users/derevandal/dev/test/dist/sw.js
url: /sw.js
resource: {
  path: '/fonts/hinted-GothamRounded-Book.woff2',
  as: 'fonts',
  crossorigin: true
}
file: C:/Users/derevandal/dev/test/dist/fonts/hinted-GothamRounded-Book.woff2
url: /fonts/hinted-GothamRounded-Book.woff2
resource: {
  path: '/fonts/hinted-GothamRounded-Bold.woff2',
  as: 'fonts',
  crossorigin: true
}
file: C:/Users/derevandal/dev/test/dist/fonts/hinted-GothamRounded-Bold.woff2
url: /fonts/hinted-GothamRounded-Bold.woff2
resource: {
  path: '/images/*.svg',
  as: 'image'
}
file: C:/Users/derevandal/dev/test/dist/images/boleto.svg
url: /images/boleto.svg
file: C:/Users/derevandal/dev/test/dist/images/cte.svg
url: /images/cte.svg
file: C:/Users/derevandal/dev/test/dist/images/cteos.svg
url: /images/cteos.svg 
file: C:/Users/derevandal/dev/test/dist/images/dots.svg 
url: /images/dots.svg
file: C:/Users/derevandal/dev/test/dist/images/esocial.svg
url: /images/esocial.svg
file: C:/Users/derevandal/dev/test/dist/images/gnre.svg 
url: /images/gnre.svg
file: C:/Users/derevandal/dev/test/dist/images/mdfe.svg 
url: /images/mdfe.svg
file: C:/Users/derevandal/dev/test/dist/images/nfce.svg 
url: /images/nfce.svg
file: C:/Users/derevandal/dev/test/dist/images/nfe.svg
url: /images/nfe.svg
file: C:/Users/derevandal/dev/test/dist/images/nfse.svg 
url: /images/nfse.svg
file: C:/Users/derevandal/dev/test/dist/images/notasegura.svg
url: /images/notasegura.svg
file: C:/Users/derevandal/dev/test/dist/images/reinf.svg
url: /images/reinf.svg 
file: C:/Users/derevandal/dev/test/dist/images/sat.svg
url: /images/sat.svg
file: C:/Users/derevandal/dev/test/dist/images/sped.svg 
url: /images/sped.svg
file: C:/Users/derevandal/dev/test/dist/images/tecnospeed.svg
url: /images/tecnospeed.svg
file: C:/Users/derevandal/dev/test/dist/images/virada.svg
url: /images/virada.svg
```

Could you check this?

Thanks

**Comment by jmblog** [comment]
@derevandal It also works on MacOS. That's great!

**Comment by jmblog** [comment]
@derevandal I've merged and released v0.1.1. Thank you for your contribution!

**Pull request merged**